fix(fontlock): fontify parameters in multiline arglist

Fuco1 opened the pull request:
Fix the anchored matcher to allow for multi-line arg list.  I will add tests if we decide to go forward with this change.

Before 
![image](https://user-images.githubusercontent.com/2664959/176743326-3cb8caf9-be40-4b75-8c44-a9a3ce743399.png)

after 

![image](https://user-images.githubusercontent.com/2664959/176743201-3aed4c51-63c2-470c-b2a2-1b68bd3db09e.png)

josteink commented:
Hey. Seems like we’re still missing some test-cases for the new syntax highlighting you’re adding support for here. 

Let me know when you think this is ready for review!

Fuco1 commented:
I added more test cases and also removed the `continued formal parameter list` rule because it is now covered by the other anchored matcher.  All existing tests pass without it as well and visually I can't tell a difference in the buffer.

josteink commented:
Simplifications are always welcome. And especially with more tests. Consider it merged!
